content/blog/an-example-application-for-processing-stock-market-trade-data-on-the-mapr-converged-data-platform.md

```diff
@@ -0,0 +1,291 @@
+
+The intent of the application is to serve as a "blueprint" for building high-speed streaming applications on the MapR Converged Data Platform.  You can use the code as a base for developing your own workflow, including producers, consumers, and analytical engines, and run queries against the indexed topics.  
+
+## Overview
+
+This project provides an engine for processing real-time streaming trading data from stock exchanges. The application consists of the following components:
+
+* A Producer microservice that streams trades using the NYSE TAQ format.  The data source is the Daily Trades dataset described [here](http://www.nyxdata.com/Data-Products/Daily-TAQ). The schema for our data is detailed in Table 6, "Daily Trades File Data Fields", on page 26 of [Daily TAQ Client Specification (from December 1st, 2013)](http://www.nyxdata.com/doc/212759).  
+* A multi-threaded Consumer microservice that indexes the trades by receiver and sender.
+* Example Spark code for querying the indexed streams at interactive speeds, enabling Spark SQL queries.  
+* Example code for persisting the streaming data to MapR-DB 
+* Performance tests for benchmarking different configurations
+* A supplementary python script to enhance the above TAQ dataset with "level 2" bid and ask data at a user-defined rate.
+
+There are several beneficial aspects of the application that are worth highlighting:
+
+* The Consumer microservice, performing the indexing, can be arbitrarily scaled simply by running more instances.  See below in this README for how to start the application.
+* Jackson annotations are provided for easy translation of the data structures to JSON and persistence to MapR-DB.
+* The application can handle 300,000 entries/second on a 3-node cluster, which is suitable for testing.  It does not require a large cluster and takes advantage of the scaling properties of MapR Streams.
+* The resulting index topics are small and can be queried fast enough such that they can be used for interactive dashboards, such as in a Zeppelin notebook.
+
+## Architecture
+
+The application provides a financial intermediary service, running *bids* and *asks* between traders.  Traders are identified with a unique ID and each bid and ask is sent from one trader to a set of N receiving traders.
+
+The following diagram shows how data moves through the architecture. The rounded rectangles represent processes that produce and/or consume data from MapR Streams topics. Java-based microservices are used to ingest and manipulate streaming data using the Kafka API. Spark and Apache Zeppelin are used to provide streaming analytics and batch-oriented visualization.
+
+<img src="https://github.com/mapr-demos/finserv-application-blueprint/raw/master/images/dataflow.gif" width="70%">
+
+## Prerequisites
+
+This application requires MapR 5.2 and Spark 2.0, which can be easily installed with the [MapR Ecosystem Pack 2.0](http://maprdocs.mapr.com/home/InteropMatrix/r_MEP_52.html).
```

Comment:
The link to MEP is not reachable/valid. MEP (MapR Ecosystem Packs) is now known as EEP (Ezmeral Ecosystem Pack) in Ezmeral Data Fabric 6.x (or later).